libs/ts-rest/core/src/lib/client.spec.ts

```diff
@@ -293,6 +293,23 @@ describe('client', () => {
         body: undefined,
       });
     });
+
+    it('w/ a non json response (string)', async () => {
```

Comment:
It's nice to have these tests anyway, but they wouldn't test this specific problem since we're mocking the api entirely, so no fetch is called or anything.